### ISSUE TITLE
packager format does not accept buildorder

### ISSUE THREAD

**Issue by mikem23** [fedora-modularity/libmodulemd#539]
I performed a simple test of using the packager format by taking a simple modulemd yaml file and changing the document value to modulemd-packager. The yaml was rejected with this error:

```
GLib.Error('Unexpected key in rpm component body: buildorder [line 34 col 9]', 'modulemd-yaml-error-quark', 8)
```

It seems like the packager format should support any field that a developer would need to specify.

**Comment by mikem23** [comment]
This makes me wonder what other necessary fields not be rejected. I haven't run exhaustive tests. Is there a clear way to compare the two formats?

**Comment by sgallagher** [comment]
> This makes me wonder what other necessary fields might be rejected. I haven't run exhaustive tests. Is there a clear way to compare the two formats?

This should actually be the only one that we dropped and it was semi-intentional.

We've been planning for the last [two years](https://pagure.io/fm-orchestrator/issue/1216) to support `buildafter` as a replacement for `buildorder`, so when I originally drafted the PackagerV2 format, I dropped `buildorder` from it. However, MBS has still not actually implemented support for `buildafter` and I overlooked that when we created the PackagerV3 format, large sections of which are shared with PackagerV2.

So we have two options here:

1) Reintroduce support for `buildorder` into PackagerV3
2) Finally implement `buildafter` in MBS.

I'd prefer that we implement `buildafter`, since it will be far simpler for packagers to understand and maintain than the `buildorder`. However, if this is impossible to fit into the schedule, we can extend PackagerV3 to support it the same way that ModuleStreamV2 did.

**Comment by mcurlej** [comment]
@sgallagher So i am confused was this removed in v2 or in v3?

**Comment by sgallagher** [comment]
It was removed in the PackagerV2 format, which we never ended up using in Fedora. The ModuleStreamV2 format still supported both `buildorder` and `buildafter`.

We then did not implement `buildorder` into the PackagerV3 format because it was not part of PackagerV2, which we were using as our baseline.

To proceed from here, we need to do one or the other of the options I described [above](https://github.com/fedora-modularity/libmodulemd/issues/539#issuecomment-758286707).

**Comment by mcurlej** [comment]
@sgallagher oh ok, thanks for the explanation. I am for the 2nd option, as `buildorder` was discontinued, some time ago.

**Comment by mcurlej** [comment]
Ok so on a quick meeting we agreed that we will add the support for `buildorder` to the v3 format. This is the best solution for now. The change in MBS would take resources and time that we don't have. According to @sgallagher to add buildorder to the v3 packager format is trivial, so this is the way to go.

**Comment by sgallagher** [comment]
I have released libmodulemd 2.12.0 including this support and submitted Fedora and EPEL 7 updates. They should be in updates-testing shortly.
